File: python/3.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: ben_cripps
# @Date:   2015-01-02 22:11:07
# @Last Modified by:   ben_cripps
# @Last Modified time: 2015-01-02 22:24:42

import math

num = 600851475143
factors = [x for x in range(1, (int (math.sqrt(num)))) if num % x == 0]

def prime(n):
    return len([x for x in range(2, n) if n % x == 0]) == 0

print(max(filter(prime, factors)))

File: c#/3.cs
using System;
using System.Collections.Generic;
using System.Linq;
                    
public class Euler
{
    private static bool isPrime(int n)
    {   
        for (var i = 2; i <= (int)Math.Sqrt(n); i++)
        {   
            if (n % i == 0)
            {   
                return false;
            }
        }
        return true;
        
    }
    
    private static object getFactors(int n) {
        List<int> factors = new List<int>();
        for (var i = 1; i < n; i++)
        {
            if (n % i == 0)
            {
                factors.Add(i);
            }
        }
        return factors;
    }

    public static void Main()
    {   
        Console.WriteLine(getFactors(100));
        Console.WriteLine(isPrime(22));
    }
    
}